indico/modules/events/contributions/controllers.py

```diff
@@ -175,3 +182,36 @@ class RHContributionPersonList(RHContributionPersonListMixin, RHManageContributi
     def _membership_filter(self):
         contribution_ids = {contrib.id for contrib in self.contribs}
         return Contribution.id.in_(contribution_ids)
+
+
+class RHContributionProtection(RHManageContributionBase):
+    """Manage contribution protection"""
+
+    def _process(self):
+        form = ContributionProtectionForm(obj=FormDefaults(**self._get_defaults()), contrib=self.contrib)
+        if form.validate_on_submit():
+            self.contrib.protection_mode = form.protection_mode.data
+            self._handle_contrib_managers(form.managers.data)
+            self._handle_access_control_list(form.acl.data)
+            self._handle_contrib_submitters(form.submitters.data)
+            return jsonify_data(flash=False)
+        return jsonify_template('events/contributions/management/contribution_protection.html', form=form,
+                                protection_message=_render_contrib_protection_message(self.contrib),
+                                contrib=self.contrib)
+
+    def _get_defaults(self):
+        managers = {p.principal for p in self.contrib.acl_entries if p.full_access}
+        submitters = {p.principal for p in self.contrib.acl_entries if p.has_management_role('submit', explicit=True)}
+        acl = {p.principal for p in self.contrib.acl_entries if p.read_access}
+        return {'managers': managers, 'submitters': submitters, 'protection_mode': self.contrib.protection_mode,
+                'acl': acl}
+
+    def _handle_contrib_managers(self, managers_data):
```

Comment:
I don't think it's really useful to have those extra methods now that they are basically one-liners